pkg/fixer/fixes/preferequalscomparison_test.go

```diff
@@ -0,0 +1,89 @@
+package fixes
+
+import (
+	"testing"
+
+	"github.com/open-policy-agent/regal/pkg/report"
+)
+
+func TestPreferEqualsComparison(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		contentAfterFix string
+		fc              *FixCandidate
+		fixExpected     bool
+		runtimeOptions  *RuntimeOptions
+	}{
+		"no change": {
+			fc:              &FixCandidate{Filename: "test.rego", Contents: "package test\n\nallow = true\n"},
+			contentAfterFix: "package test\n\nallow = true\n",
+			fixExpected:     false,
+			runtimeOptions:  &RuntimeOptions{},
+		},
+		"no change because no location": {
+			fc:              &FixCandidate{Filename: "test.rego", Contents: "package test\n\nallow = true\n"},
+			contentAfterFix: "package test\n\nallow = true\n",
+			fixExpected:     false,
+			runtimeOptions:  &RuntimeOptions{},
+		},
+		"single change": {
+			fc:              &FixCandidate{Filename: "test.rego", Contents: "package test\n\nallow := true\nallow = true\n"},
+			contentAfterFix: "package test\n\nallow := true\nallow == true\n",
+			fixExpected:     true,
+			runtimeOptions:  &RuntimeOptions{Locations: []report.Location{{Row: 4, Column: 7}}},
+		},
+		"bad change": {
+			fc:              &FixCandidate{Filename: "test.rego", Contents: "package test\n\nallow := true\nallow = true\n"},
+			contentAfterFix: "package test\n\nallow := true\nallow = true\n",
+			fixExpected:     false,
+			runtimeOptions:  &RuntimeOptions{Locations: []report.Location{{Row: 1, Column: 1}}},
+		},
+		"many changes": {
+			fc: &FixCandidate{
+				Filename: "test.rego",
+				Contents: "package test\n\nallow := true\n\nallow = true\nallow = false\n",
+			},
+			contentAfterFix: "package test\n\nallow := true\n\nallow == true\nallow == false\n",
+			fixExpected:     true,
+			runtimeOptions:  &RuntimeOptions{Locations: []report.Location{{Row: 5, Column: 7}, {Row: 6, Column: 7}}},
+		},
+		"different columns": {
+			fc: &FixCandidate{
+				Filename: "test.rego",
+				Contents: "package test\n\nlong := true\n\nlonger = true\nlongest = false\n",
+			},
+			contentAfterFix: "package test\n\nlong := true\n\nlonger == true\nlongest == false\n",
```

Comment:
I think for some of these tests we might use a raw string instead.
```suggestion
			contentAfterFix: `package test
long := true

longer == true

longest == false
`,
```